### ISSUE TITLE
Procedures for running ThermalZones.EnergyPlus models should be improved

### ISSUE THREAD

**Issue by casella** [lbl-srg/modelica-buildings#4055]
The current version of Buildings on the master branch contains these two packages:
- Buildings.ThermalZones.EnergyPlus_9_6_0
- Buildings.ThermalZones.EnergyPlus_24_2_0

which depend on some pre-compiled spawn executable, through external objects. How to reliably and consistently get these two packages to work across platform is not entirely clear.

### On Linux

The OSMC is running tests on its CI under Linux, see e.g. https://libraries.openmodelica.org/branches/master/Buildings_latest/Buildings_latest.html

In order to get the EnergyPlus_9_6_0 tests going, the test script was set up to call a Python script `$resourceLocation/src/ThermalZones/install.py`. As we see from the source code
https://github.com/lbl-srg/modelica-buildings/blob/07dfbd56db0f683827470f7ebf77e2248adaf19e/Buildings/Resources/src/ThermalZones/install.py#L12
it seems that it installs both versions of spawn, one required for EP 9.6.0 and one required for EP 24.2.0

Unfortunately, we couldn't find the documentation about installing the spawn libraries using install.py in the user's guide of both packages.

Q1: is this python script still valid and maintained, and does it work for both 9.6.0 and 24.2.0?
Q2: if so, why isn't it documented?

The current documentation asks for adding stuff to the system-wide PATH. IMHO this is something that should be avoided, one should not pollute the system PATH with such things.

Q3: is it really necessary to change the PATH, or is is sufficient to copy the spawn executables in some proper places? 

By looking at the "How is spawn invoked?" section of the documentation, it seems that this should be the case, but the installation instructions don't mention this option explicitly

### On Windows

On windows, python is not installed by default. In fact, the documentation doesn't mention the python installer at all, so this is OK on this OS, apparently. The other questions apply

Q4: Is it really necessary to change the path?
Q5: Is it sufficient to place the .exe in the proper place, as implicitly suggested by the "How is spawn invoked" section?

We actually tried that, but it didn't work, further adding confusion and frustration 😅

**Comment by casella** [comment]
If you use the loadResources() function and the Modelica URIs, it is possible to get the absolute path of the binaries if they are installed in some pre-specified place under the Buildings/Resources directory.

Hence, I think it should be possible to avoid mangling the PATH entirely. To me, this is a much more preferrable solution, because it works on a local level and avoids messing up stuff at the system level.

**Comment by AndreaBartolini** [comment]
As additional information: I've python installed on my Win11-Pro PC (because I use it for my job in general), by running the `$resourceLocation/src/ThermalZones/install.py` script the external Energy Plus libraries (both the 9.6.0 and the 24.2.0)  work fine on Win11-Pro.

My Path os-variables don't contain the path to the Energy Plus executables (@casella I've deleted the old references in my path variables).

**Comment by mwetter** [comment]
There is no need to update the `PATH` variable. If the Buildings library is cloned from github, just run 
```
Buildings/Resources/src/ThermalZones/install.py --binaries-for-os-only
```
(see also https://github.com/lbl-srg/modelica-buildings/blob/master/README.md).
This will install the binaries in subfolders of the `Resources` folder.

If the installation is using the zip file from an official release on https://github.com/lbl-srg/modelica-buildings/releases then all binaries are in the zip file.

The installation is also described in https://simulationresearch.lbl.gov/modelica/releases/v11.0.0/help/Buildings_ThermalZones_EnergyPlus_9_6_0_UsersGuide.html#Buildings.ThermalZones.EnergyPlus_9_6_0.UsersGuide.Installation
although that page does not describe `install.py` which was added later and didn't make it into this description.

The option with modifying the `PATH` is for those who may have a separate installation of the binaries outside of the `Buildings` library which was a use case we needed to support. The `PATH` is only used as a fallback if the local installation in `Resources` is missing.

I will update the documentation to make this clearer.
